### ISSUE TITLE
Guidance site components page heading structure doesn't follow logical order

### ISSUE THREAD

**Issue by DBD324** [mi6/ic-design-system#611]
## Summary of the bug

On components pages, the heading structure doesn't follow the logical heading order. 

We have:
App title H1
Page header title H2
First content heading H4
Interactive example H3

## 🪜 How to reproduce

Tell us the steps to reproduce the problem:

1. Go to page: Buttons component page
2. Launch NVDA/Voiceover
3. Use 'H' key to navigate through headings on page
4. See the error

(Alternatively use a tool like WAVE)
![Screenshot 2023-11-02 at 12 22 11](https://github.com/mi6/ic-design-system/assets/111292477/de829658-fe08-4196-9de1-3e79cdf9022f)

## 📸 Screenshots or code

If you can, add screenshots to show the problem. If you need to show a code example, you can include a snippet or link to a repository with an example.

## 🖥 📱 Device

- Type: All
- Device: All
- OS version: All
- Browser version: All

## 🧐 Expected behaviour

Change top level headings on page content to use H3, instead of H4, as this follows the logical structure. 

E.g on buttons page. 

H1 = ICDS
H2 = Buttons (page header title)
H3 = Component Variants (first main content heading)
H3 = When to use
H4 = Primary buttons

For interactive components, investigate whether they need a heading. If so, make sure that they use a heading level one lower than the heading of the section they sit in. 
For example, a component demo sits under a H3, therefore its heading should be H4. 
Or, a component demo sits under a H4, therefore its heading should be H5. 

## 📝 Acceptance Criteria

If relevant, describe in full detail the different interactions and edge cases that the component or patterns needs to fulfil.

_Given_ that a components page exists
_When_ navigating through headings on the page
_Then_ the headings should follow a logical heading structure. 

## Additional info

Tell us anything else useful to help us fix or understand the problem.
